06-r-and-sql.Rmd

```diff
@@ -0,0 +1,144 @@
+---
+title: "SQL databases and R"
+output:
+  html_document:
+    toc: true
+    theme: united
+---
+
+
+Learning goals
+------
+Students should finish the lesson with the following skills.
+
+* Students will be able to connect to an existing database using R
+* Query an existing database through R using queries constructed with variables
+* Create a database and populate it from R
+* Understand that scripted database interactions enhance reproducibility of analysis
+
+
+Introduction
+---------
+
+Interacting with databases through scripted languages can offer advantages over querying databases via a GUI interface.  A GUI interfaces for your database is easier to use and allows the quick viewing of adhoc queries.  Querying a database with a programatic intreface (in this case R, but it could be any language) are slightly more complicated.  However the trade-off is that data manipulations are preserved in the code.  Aggregations, summaries and other database operations are preserved.  Therefore those pre-analysis data manipulation steps are not lost.
+
+Connecting R to sqlite databases
+---------
+
+R can connect to databases through a number of libraries.  In our case we we will use [RSQLite](http://cran.r-project.org/web/packages/RSQLite/index.html) to connect to existing SQLite3 databases.  However you should be able to connect on almost any database in R via JDBC[http://cran.r-project.org/web/packages/RJDBC/index.html](http://cran.r-project.org/web/packages/RJDBC/index.html) or [ODBC](http://cran.r-project.org/web/packages/RODBC/index.html), or specific database libraries (such as we are doing, or [MySQL](http://cran.r-project.org/web/packages/RMySQL/index.html) ).  
+
+To begin these exercises we'll connect to the mammal database you've just created.
+
+```{r connect}
+library(RSQLite)
+## Set dbname and driver out of convenience
+myDB <- "~/Downloads/portal_mammals.sqlite"
```

Comment:
maybe just the file name here? as people may not have downloaded it in their download folder.